### PR TITLE
handle relation table name customization

### PR DESCRIPTION
**Pull request by ZeBigDuck**
I updated configuration file to be able to handle relation table names, resolving issue https://github.com/dunglas/php-schema/issues/17

### CODE REVIEW

File: src/SchemaOrgModel/TypesGenerator.php [hunk 1]
```diff
@@ -236,6 +236,11 @@ public function generate($config)
                     }
                 }
 
+                $relationTableName = null;
```
Comment (dunglas) [inline src/SchemaOrgModel/TypesGenerator.php]: Isn't it better to move this logic in the `DoctrineOrmAnnotationGenerator`?

File: src/SchemaOrgModel/AnnotationGenerator/DoctrineOrmAnnotationGenerator.php [hunk 2]
```diff
@@ -140,16 +140,27 @@ public function generateFieldAnnotations($className, $fieldName)
...
+                    } else {
+                        $annotations[] = '@ORM\JoinTable(inverseJoinColumns={@ORM\JoinColumn(unique=true)})';
+                    }
                     break;
 
                 case CardinalitiesExtractor::CARDINALITY_1_N:
                     $annotations[] = sprintf('@ORM\ManyToMany(targetEntity="%s")', $this->getRelationName($field['range']));
-                    $annotations[] = '@ORM\JoinTable(inverseJoinColumns={@ORM\JoinColumn(nullable=false, unique=true)})';
+                    if ($field['relationTableName']) {
+                        $annotations[] = sprintf('@ORM\JoinTable(name="%s", inverseJoinColumns={@ORM\JoinColumn(nullable=false, unique=true)})', $field['relationTableName']);
```
Comment (dunglas) [inline src/SchemaOrgModel/AnnotationGenerator/DoctrineOrmAnnotationGenerator.php]: It is better to not duplicate the code here. What about something like the following:

```php
$name = $field['relationTableName'] ? sprintf('name="%s", ', $field['relationTableName']) : '';

$annotations[] = '@ORM\JoinTable(' . $name . 'inverseJoinColumns={@ORM\JoinColumn(nullable=false, unique=true)})';
```

**Comment by dunglas** [comment]
:+1:

I left some comments. Can you also update tests?

**Comment by ZeBigDuck** [comment]
I updated tests and moved logic to where it's expected to be.

I'm not used to deal with tests (booh), I hope I did what you did expect!

**Comment by dunglas** [comment]
Can you squash your commits?

**Comment by ZeBigDuck** [comment]
This should be ready to merge!

**Comment by dunglas** [comment]
Yes! Can you just squash your commits?

**Comment by ZeBigDuck** [comment]
Damned...

It seems I squashed... but one of your commits look to have been part of it...

**Comment by dunglas** [comment]
Merged (67492ed23e9eb90d77f836d592617665dfa23dc2). Thank you @ZeBigDuck

**Pull request closed without merging**